### PR TITLE
feat: password reset

### PR DESCRIPTION
**Pull request by KubaBaniak**
# [Reset password](https://app.clickup.com/t/8692nwt7u)

## Acceptance criteria:
- User request an email with reset password link
- Token is valid for 5 mins 24h
- Do not send response, that emails exists and has been sent

### CODE REVIEW

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -75,4 +92,43 @@ export class AuthController {
...
+  @Get('activate-account')
+  async activateAccount(@Query('token') token: string): Promise<void> {
+    const tokenData = await this.authService.verifyAccountActivationToken(
+      token,
+    );
+    await this.authService.activateAccount(tokenData.id);
+  }
+
+  @ApiOperation({ summary: 'Sends link to resets password of the user' })
+  @Get('reset-password-email')
```
Comment (rad20c) [inline src/auth/auth.controller.ts:106]: hmm maybe something like `request-password-reset`? Or `reset-password`, and the next one should be then `change-password`?

File: src/auth/test/auth.controller.spec.ts [hunk 2]
```diff
@@ -83,10 +102,57 @@ describe('AuthController', () => {
...
+      jest.spyOn(mailService, 'sendResetPasswordEmail');
+
+      await authController.resetPasswordEmail(request);
+
+      expect(authService.generateResetPasswordToken).toHaveBeenCalled();
+      expect(mailService.sendResetPasswordEmail).toHaveBeenCalled();
+    });
+  });
+
+  it('should change password for the provided one via form', async () => {
```
Comment (rad20c) [inline src/auth/test/auth.controller.spec.ts:140]: I think `should change password` should be enough. The new password is not really provided in the form (as form is on the frontend side) but in a request body

File: src/mail/mail.service.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  async sendAccountActivationEmail(
+    userEmail: string,
+    token: string,
+  ): Promise<SentMessageInfo> {
+    const url = `${process.env.BASE_URL}/auth/activate-account/?token=${token}`;
+
+    return this.mailerService.sendMail({
+      to: userEmail,
+      subject: 'Welcome to Nice App! Confirm your Email',
```
Comment (rad20c) [inline src/mail/mail.service.ts:17]: Recipe App 😅

File: src/mail/templates/account-activation-email.pug [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  head
+    meta(charset="UTF-8")
+    meta(name="viewport", content="width=device-width, initial-scale=1.0")
+    title Recipe App - Account Activation
+  body(style="font-family: Arial, sans-serif; background-color: #f5f5f5; padding: 20px; text-align: center;")
+    table(style="max-width: 600px; margin: 0 auto; background-color: #ffffff; border-radius: 10px; box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);")
+      tr
+        td(style="padding: 20px;")
+          h1(style="font-size: 24px; color: #333;") Recipe App Account Activation
+          p(style="font-size: 16px; color: #666; margin-top: 10px;") Welcome to Recipe App! To get started, please activate your account by clicking the button below:
```
Comment (rad20c) [inline src/mail/templates/account-activation-email.pug:12]: why #666? 😅

File: src/main.ts [hunk 5]
```diff
@@ -18,6 +18,8 @@ async function bootstrap() {
     }),
   );
 
+  app.enableCors();
```
Comment (rad20c) [inline src/main.ts:21]: just a question: why did you need to enable CORS?

**Review by rad20c — APPROVED** [review]

File: src/main.ts (hunk 5 above)
Comment (KubaBaniak) [reply src/main.ts:21]: `Access to fetch at 'http://localhost:3000/auth/reset-password' from origin 'null' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`
